third_party/libunwindstack/tests/PeCoffUnwindInfoUnwinderX86_64Test.cpp

```diff
@@ -0,0 +1,385 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PeCoffUnwindInfoUnwinderX86_64.h"
+
+#include <algorithm>
+#include <limits>
+
+#include "unwindstack/Error.h"
+#include "unwindstack/MachineX86_64.h"
+#include "unwindstack/RegsX86_64.h"
+#include "utils/MemoryFake.h"
+
+#include "PeCoffEpilog.h"
+#include "PeCoffRuntimeFunctions.h"
+#include "PeCoffUnwindInfoEvaluator.h"
+#include "PeCoffUnwindInfos.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+namespace unwindstack {
+
+class MockPeCoffRuntimeFunctions : public PeCoffRuntimeFunctions {
+ public:
+  MockPeCoffRuntimeFunctions() = default;
+
+  MOCK_METHOD(bool, Init, (uint64_t, uint64_t), (override));
+  MOCK_METHOD(bool, FindRuntimeFunction, (uint64_t, RuntimeFunction*), (const, override));
+};
+
+class MockPeCoffUnwindInfos : public PeCoffUnwindInfos {
+ public:
+  MockPeCoffUnwindInfos() = default;
+
+  MOCK_METHOD(bool, GetUnwindInfo, (uint64_t, UnwindInfo*), (override));
+};
+
+class MockPeCoffEpilog : public PeCoffEpilog {
+ public:
+  MockPeCoffEpilog() = default;
+
+  MOCK_METHOD(bool, Init, (), (override));
+  MOCK_METHOD(bool, DetectAndHandleEpilog, (uint64_t, uint64_t, uint64_t, Memory*, Regs*),
+              (override));
+
+  void FailWithError(ErrorCode error_code) { last_error_.code = error_code; }
+};
+
+class MockPeCoffUnwindInfoEvaluator : public PeCoffUnwindInfoEvaluator {
+ public:
+  MockPeCoffUnwindInfoEvaluator() = default;
+
+  MOCK_METHOD(bool, Eval, (Memory*, Regs*, const UnwindInfo&, PeCoffUnwindInfos*, uint64_t),
+              (override));
+};
+
+class TestPeCoffUnwindInfoUnwinderX86_64 : public PeCoffUnwindInfoUnwinderX86_64 {
+ public:
+  TestPeCoffUnwindInfoUnwinderX86_64()
+      : PeCoffUnwindInfoUnwinderX86_64(nullptr, 0, 0, 0, 0, 0, {}) {}
+
+  void SetFakeRuntimeFuntions(std::unique_ptr<MockPeCoffRuntimeFunctions> runtime_functions) {
+    this->runtime_functions_ = std::move(runtime_functions);
+  }
+
+  void SetFakeUnwindInfos(std::unique_ptr<MockPeCoffUnwindInfos> unwind_infos) {
+    this->unwind_infos_ = std::move(unwind_infos);
+  }
+
+  void SetFakeEpilog(std::unique_ptr<MockPeCoffEpilog> epilog) {
+    this->epilog_ = std::move(epilog);
+  }
+
+  void SetFakeUnwindInfoEvaluator(
+      std::unique_ptr<MockPeCoffUnwindInfoEvaluator> unwind_info_evaluator) {
+    this->unwind_info_evaluator_ = std::move(unwind_info_evaluator);
+  }
+};
+
+TEST(PeCoffUnwindInfoUnwinderX86_64Test, step_succeeds_on_leaf_functions) {
+  TestPeCoffUnwindInfoUnwinderX86_64 fake_unwinder;
```

Comment:
I just realized: you could rename these variables too

Reply:
Yeah, should have seen this myself. Thanks for pointing it out. Fixed.